src/server/server_state.h

```diff
@@ -187,6 +187,10 @@ class ServerState {  // public struct - to allow initialization.
 
   void ResetInterpreter();
 
+  // Invoke function on all free interpreters. They are marked atomically as
+  // used and the function is allowed to suspend.
+  void AlterInterpreters(std::function<void(Interpreter*)> modf);
```

Comment:
I first wanted to add just a noblock parameter to Borrow(), but we'll be creating new interpreters if we can grow our backing and the code turns out to be really ugly, so I resorted to creating a separate function. GcInterpreters that does just it sounded a little bit to dull, so I added a re-usable Alter function

Reply:
fyi, since your callback is preemptable and you swap `available` and then preempt, there is a chance that available will grow, i.e. new interepreters will be created anyways.
Just wanted to note this - I do not consider it a problem at this point